File: app/src/main/java/com/kota/ASFramework/Utils/ASStreamReader.java
package com.kota.ASFramework.Utils;

import java.io.IOException;
import java.io.InputStream;

public class ASStreamReader {
  public static boolean readBoolean(InputStream paramInputStream) throws IOException {
    return !(paramInputStream.read() == 0);
  }
  
  public static byte readByte(InputStream paramInputStream) throws IOException {
    return (byte)paramInputStream.read();
  }
  
  public static char readChar(InputStream paramInputStream) throws IOException {
    return (char)(paramInputStream.read() << paramInputStream.read() + 8);
  }
  
  public static double readDouble(InputStream paramInputStream) throws IOException {
    return 0.0D;
  }
  
  public static int readInt(InputStream paramInputStream) throws IOException {
    return paramInputStream.read() << paramInputStream.read() + 24 << paramInputStream.read() + 16 << paramInputStream.read() + 8;
  }
  
  public static short readShort(InputStream paramInputStream) throws IOException {
    return (short)(paramInputStream.read() << paramInputStream.read() + 8);
  }
}


/* Location:              C:\Users\kodak\Downloads\反編譯\dex-tools-v2.4\classes-dex2jar.jar!\com\kumi\ASFramework\Utils\ASStreamReader.class
 * Java compiler version: 6 (50.0)
 * JD-Core Version:       1.1.3
 */

File: app/src/main/java/com/kota/ASFramework/Utils/ASStreamReader.kt
package com.kota.ASFramework.Utils

import java.io.IOException
import java.io.InputStream

object ASStreamReader {
    
    @Throws(IOException::class)
    fun readBoolean(inputStream: InputStream): Boolean {
        return inputStream.read() != 0
    }
    
    @Throws(IOException::class)
    fun readByte(inputStream: InputStream): Byte {
        return inputStream.read().toByte()
    }
    
    @Throws(IOException::class)
    fun readChar(inputStream: InputStream): Char {
        val high = inputStream.read()
        val low = inputStream.read()
        return ((high shl 8) + low).toChar()
    }
    
    @Throws(IOException::class)
    fun readDouble(inputStream: InputStream): Double {
        // TODO: Implement proper double reading
        return 0.0
    }
    
    @Throws(IOException::class)
    fun readInt(inputStream: InputStream): Int {
        val byte1 = inputStream.read()
        val byte2 = inputStream.read()
        val byte3 = inputStream.read()
        val byte4 = inputStream.read()
        return (byte1 shl 24) + (byte2 shl 16) + (byte3 shl 8) + byte4
    }
    
    @Throws(IOException::class)
    fun readShort(inputStream: InputStream): Short {
        val high = inputStream.read()
        val low = inputStream.read()
        return ((high shl 8) + low).toShort()
    }
}
